src/main/java/com/three_tech_solutions/slot_app/services/implementations/SlotServiceImpl.java

```diff
@@ -37,6 +44,36 @@ public void createSlot(CreateSlotRequest request) {
         slotRepository.save(buildSlot(request));
     }
 
+    @Override
+    public ListSlotsResponse getSlotsByDayOfWeek(UUID userId, DayOfWeek dayOfWeek) {
+
+        List<Slot> slots = slotRepository.findAllByUserIdAndDayOfWeekOrdered(userId, dayOfWeek);
+
+        List<SlotResponse> responses = slots.stream()
+                .map(slot -> {
+                    return slotMapper.toSlotResponse(slot, calculateUsedCapacity(slot));
+                })
+                .toList();
+
+        return new ListSlotsResponse(responses.size(), responses);
+    }
```

Comment:
Esto quizas lo podemos refactorizar asi, primero armas estos metodos

```
    private List<Slot> getSlotsByUserAndDayOfWeek(UUID userId, DayOfWeek dayOfWeek) {
        return slotRepository.findAllByUserIdAndDayOfWeekOrdered(userId, dayOfWeek);
    }
```

```
    private Collector<SlotResponse, Object, ListSlotsResponse> collectListAndBuildListSlotsResponse() {
        return Collectors.collectingAndThen(
                Collectors.toList(),
                list -> new ListSlotsResponse(list.size(), list)
        );
    }
```

y esto lo modificas asi


```suggestion
    @Override
    public ListSlotsResponse getSlotsByDayOfWeek(UUID userId, DayOfWeek dayOfWeek) {
        return getSlotsByUserAndDayOfWeek(userId, dayOfWeek).stream()
                .map(slot -> slotMapper.toSlotResponse(slot, calculateUsedCapacity(slot)))
                .collect(collectListAndBuildListSlotsResponse());
    }
```